src/configreader.cpp

```diff
@@ -347,7 +347,7 @@ void ServerConfig::Fill()
 	}
 	SoftLimit = ConfValue("performance")->getUInt("softlimit", (SocketEngine::GetMaxFds() > 0 ? SocketEngine::GetMaxFds() : LONG_MAX), 10);
 	CCOnConnect = ConfValue("performance")->getBool("clonesonconnect", true);
-	MaxConn = ConfValue("performance")->getUInt("somaxconn", SOMAXCONN);
+	MaxConn = static_cast<unsigned int>(ConfValue("performance")->getUInt("somaxconn", SOMAXCONN, 1, SOMAXCONN));
```

Comment:
Is this valid? You can set net.core.somaxconn higher than SOMAXCONN (4096 on my system) with sysctl.